[TS] add production deploy checklist (#719)
temporalio/documentation: docs/content/how-to-develop-a-worker-program-in-typescript.md

Workers are very flexible - you can host any or all of your Workflows and Activities on a Worker, and you can host multiple Workers in a single machine.

There are three main things the Worker needs:

- `taskQueue`: the Task Queue to poll. This is the only required argument.

> - `activities`: You import and supply these directly to the Worker (optional if no)

Author: ```suggestion
- `activities`: Optional. Imported and supplied directly to the Worker. Not file path name. 
```

> - `activities`: You import and supply these directly to the Worker (optional if no )

Reviewer: `if no )`

Author: ```suggestion
- `activities`: You import and supply these directly to the Worker (optional if no)
```
